Jellyfin.Server.Implementations/FullSystemBackup/BackupService.cs

```diff
@@ -0,0 +1,437 @@
+        _applicationHost.NotifyPendingRestart();
+    }
+
+    /// <inheritdoc/>
+    public async Task RestoreBackupAsync(string archivePath)
+    {
+        _logger.LogWarning("Begin restoring system to {BackupArchive}", archivePath); // Info isn't cutting it
+        if (!File.Exists(archivePath))
+        {
+            throw new FileNotFoundException($"Requested backup file '{archivePath}' does not exist.");
+        }
+
+        var fileStream = File.OpenRead(archivePath);
+        await using (fileStream.ConfigureAwait(false))
+        {
+            using var zipArchive = new ZipArchive(fileStream, ZipArchiveMode.Read, false);
+            var zipArchiveEntry = zipArchive.GetEntry(ManifestEntryName);
+
+            if (zipArchiveEntry is null)
+            {
+                throw new NotSupportedException($"The loaded archive '{archivePath}' does not appear to be a Jellyfin backup as its missing the '{ManifestEntryName}'.");
+            }
+
+            BackupManifest? manifest;
+            var manifestStream = zipArchiveEntry.Open();
+            await using (manifestStream.ConfigureAwait(false))
+            {
+                manifest = await JsonSerializer.DeserializeAsync<BackupManifest>(manifestStream, _serializerSettings).ConfigureAwait(false);
+            }
+
+            if (manifest!.ServerVersion > _applicationHost.ApplicationVersion) // newer versions of Jellyfin should be able to load older versions as we have migrations.
+            {
+                throw new NotSupportedException($"The loaded archive '{archivePath}' is made for a newer version of Jellyfin ({manifest.ServerVersion}) and cannot be loaded in this version.");
+            }
+
+            if (!TestBackupVersionCompatibility(manifest.BackupEngineVersion))
+            {
+                throw new NotSupportedException($"The loaded archive '{archivePath}' is made for a newer version of Jellyfin ({manifest.ServerVersion}) and cannot be loaded in this version.");
+            }
+
+            static async Task CopyOverride(ZipArchiveEntry item, string targetPath)
+            {
+                var targetStream = File.Create(targetPath);
+                await using (targetStream.ConfigureAwait(false))
+                {
+                    var sourceStream = item.Open();
+                    await using (sourceStream.ConfigureAwait(false))
+                    {
+                        await sourceStream.CopyToAsync(targetStream).ConfigureAwait(false);
+                    }
+                }
+            }
+
+            async Task CopyDirectory(string source, string target)
+            {
+                if (!Directory.Exists(source))
+                {
+                    Directory.CreateDirectory(source);
+                }
+
+                var configFiles = zipArchive.Entries
+                    .Where(e => Path.GetFullPath(e.FullName.TrimEnd(Path.DirectorySeparatorChar) + Path.DirectorySeparatorChar)
```

Comment:
```suggestion
                    .Where(e => Path.GetFullPath(Path.Combine(target, e.FullName.TrimEnd(Path.DirectorySeparatorChar) + Path.DirectorySeparatorChar))
```

I think the Path.Combine is missing, the entry full name will likely never start with the target path

Reply:
For example target is (from first invocation) `Config/`
and entries are stored at L325 for example as `Config/users`
So they should always match